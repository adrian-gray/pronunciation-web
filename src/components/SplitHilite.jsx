import React from "react";

import Typography from "@material-ui/core/Typography";

const styles = theme => ({
  hilite: theme.hilite
});

export function SplitHilite(props) {
  const { classes, str } = props;
  const elements = str.split(/(~)/);

  const arr = [];
  let index = 0;
  do {
    let el = elements.shift();
    if (el === "~") {
      const txt = elements.shift();
      arr.push(
        <span className={classes.hilite} key={`${txt}${index}`}>
          {txt}
        </span>
      );
      elements.shift(); // closing ~
    } else {
      arr.push(<span key={`${el}${index}`}>{el}</span>);
    }
    index++;
  } while (elements.length);
  arr.push(" ");

  return arr;
}

export default withStyles(styles)(SplitHilite);
