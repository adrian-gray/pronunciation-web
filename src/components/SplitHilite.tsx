import React from "react";

import { makeStyles } from "@material-ui/core/styles";
import { withTheme } from "@material-ui/styles";

const useStyles = makeStyles(theme => ({
  hilite: theme.hilite
}));

const SplitHilite = props => {
  const classes = useStyles(props);

  const { str } = props;
  const elements = str.split(/(~)/);
  const arr = [];
  let index = 0;

  do {
    let el = elements.shift();
    if (el === "~") {
      const txt = elements.shift();
      arr.push(
        <span className={classes.hilite} key={`${txt}${index}`}>
          {txt}
        </span>
      );
      elements.shift(); // closing ~
    } else {
      arr.push(<span key={`${el}${index}`}>{el}</span>);
    }
    index++;
  } while (elements.length);
  arr.push(" ");

  return arr;
};

export default withTheme(SplitHilite);
