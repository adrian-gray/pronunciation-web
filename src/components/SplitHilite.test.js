import React from "react";
import { render, cleanup } from "@testing-library/react";

import SplitHilite from "./SplitHilite";

afterEach(cleanup);

test("SplitHilite", async () => {
  const { container } = render(<SplitHilite str="pl~i~zzz~i~sed" />);
  expect(container.firstChild).toMatchSnapshot();
});
