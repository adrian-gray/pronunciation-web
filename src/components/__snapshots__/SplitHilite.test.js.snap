// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplitHilite 1`] = `
<div>
  <span>
    pl
  </span>
  <span
    class="hilite"
  >
    i
  </span>
  <span>
    zzz
  </span>
  <span
    class="hilite"
  >
    i
  </span>
  <span>
    sed
  </span>
   
</div>
`;
